gframe/duelclient.cpp

```diff
@@ -600,7 +600,12 @@ void DuelClient::HandleSTOCPacketLanAsync(const std::vector<uint8_t>& data) {
 		break;
 	}
 	case STOC_SELECT_HAND: {
-		mainGame->wHand->setVisible(true);
+		if (mainGame->gSettings.chkAutoRPS->isChecked()) {
+			std::lock_guard lock(mainGame->gMutex);
+			mainGame->SendRPSResult(std::uniform_int_distribution<>(1, 3)(rnd));
+		} else {
+			mainGame->wHand->setVisible(true);
+		}
```

Comment:
the existing code didn't lock on the mutex when showing the RPS frame, so I don't either... sending the result without the lock will tend to hang the client, though.

`DuelClient::ClientAnalyze` always locks the mutex. Should this lock the mutex when showing the RPS frame?